### PR TITLE
Split AgreementList out from UserProfile.

### PR DESCRIPTION
**Pull request by kadams54**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5495/
Test FAILed.

File: jujugui/static/gui/src/app/components/user-profile/agreement-list.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('agreement-list', function() {
+
+  juju.components.AgreementList = React.createClass({
+    xhrs: [],
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/agreement-list.js]: This should be defined in the `getInitialState` method as `this.activeXhr = []` as right now it's shared by all instances of `AgreementList`. Which isn't likely an issue, but a good practice to get into.

File: jujugui/static/gui/src/app/components/user-profile/agreement-list.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        );
+      }
+      var list = this.state.agreementList;
+      if (!list || list.length === 0) {
+        return null;
+      }
+      var rows = [];
+      list.forEach((model) => {
+        rows.push(this._generateRow(model));
+      });
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/agreement-list.js]:
```JavaScript
let rows = list.map(this._generateRow);
```

**Comment by hatched** [comment]
I see tests added, but none removed from the old suite, was it just poorly tested before?

**Comment by kadams54** [comment]
I'm not sure it was poorly tested per se, but everything was all tested together. So "render an empty state" tested all of the lists, and "render entities" tested all of the lists... so on and so forth. Consequently it was more a matter of replacing the rendered output with the new AgreementList component in existing tests.

File: jujugui/static/gui/src/app/components/user-profile/agreement-list.js (hunk 2 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/agreement-list.js]: Very nice, but I want a branch that I can land without breaking CI :-\

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5499/
Test PASSed.

File: jujugui/static/gui/src/app/components/user-profile/agreement-list.js (hunk 2 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/agreement-list.js]: Never mind, ignore me.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5500/
Test PASSed.

**Comment by hatched** [comment]
QA Issue:
I believe that all of the following issues are not caused by your branch so QA :ok:

The first time visiting the profile page the spinner for the agreementlist stayed spinning for quite some time. After which the `_makeRequest` failure handler had a syntax error `var data = JSON.parse(xhr.target.responseText);` because `xhr="Unexpected end of JSON input"` https://github.com/juju/juju-gui/issues/1882

The second time I was shown the "your profile is currently empty" even though I have agreements

the third time the request to agreements was cancelled

**Comment by kadams54** [comment]
Thanks! Yeah, this PR's intent is to better organize the code, making it easier to fix all the brokeness in subsequent PRs.

**Comment by kadams54** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**